src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState(Status.all);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filterTodos = (todosToFilter: Todo[], status: Status, searchTerm: string): Todo[] => {
+    if (status === Status.all && !searchTerm.trim()) {
+      return todosToFilter;
+    }
+
+    return todosToFilter.filter(currentTodo => {
+      if (status === Status.active && currentTodo.completed) {
+        return false;
+      }
+
+      if (status === Status.completed && !currentTodo.completed) {
+        return false;
+      }
+
+      if (searchTerm.trim()
+        && !currentTodo.title.toLowerCase()
+          .includes(searchTerm.toLowerCase())
+      ) {
+        return false;
+      }
+
+      return true;
+    });
+  };
```

Comment:
My assumption is spli this on 2 cases:
Why do you need need filter by query in case query is empt or why do you do you need filter by status in case query is not empty but is `Status.all`